torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -6916,6 +6916,26 @@ mkldnn_rnn_layer_differentiable_backward(
   return std::make_tuple(cat_layer_dx, dWx, dWh, db, db, dprev_h, dprev_c);
 }
 
+std::tuple<Tensor, Tensor, Tensor> attn_backward(
+    const variable_list& grads,
+    const Tensor& query,
+    const Tensor& key,
+    const Tensor& value) {
+  // FIXME Is there a way to pass x and attn without computing them here?
+  // Maybe I can also return x in forward function and access attn, x via
+  // result1, result2?
```

Comment:
You can indeed, either name your results and then use this name in the formula.
Or use result0 / result1 which are the default names if you didn't set any.

Reply:
updated it to use result in 83918e7655